File: push-client-python/auto_confirm_message_handler.py

import json
from client_message import *
from client import *
class AutoConfirmMessageHandler():

  def __init__(self, callback, client: Client):
    self.callback = callback
    self.client = client

  def handle(self, msg):
    try:
      maps = json.loads(msg)
      clientMessage = ClientMessage()
      clientMessage.messageId = maps['messageId']
      clientMessage.messageType = maps['messageType']
      clientMessage.payload = maps['payload']
      self.callback(clientMessage)
      self.client.send('confirm-' + clientMessage.messageId)
    except Exception as e:
      self.__handleUnStructMsg__(msg)
  
  def __handleUnStructMsg__(self, msg):
    if msg.startswith('kickout'):
      print('被踢出!!!')
      self.client.close()
    else:
      print ('接收到非结构化消息:' + msg)

File: push-clients/push-client-javascript/auto_confirm_message_handler.js

class AutoConfrimMessageHandler{
  constructor(callback, client){
    this.callback = callback
    this.client = client
  }

  handle(msg){
    try {
      const maps = JSON.parse(msg)
      this.callback(maps)
      this.client.send('confirm-' + maps.messageId)
    } catch(e){
      this.handleUnStructMsg(msg)
    }
  }

  handleUnStructMsg(msg){
    if (msg.startsWith('kickout')){
      console.log('被踢出！！！')
      this.client.close()
    }else{
      console.log('接收到非结构化消息:', msg)
    }
  }
}

module.exports = { AutoConfrimMessageHandler }